### PR TITLE
Adding custom views as Empty State

### PR DESCRIPTION
**Pull request by lojals**
#### What does this PR do? Any background context you want to provide?
This PR allows users to use custom views as empty states
#### Where should the reviewer start?
* `WLEmptyState/Classes/Protocols/WLEmptyStateDataSource.swift`
* `WLEmptyState/Classes/Protocols/WLEmptyStateProtocol.swift`
#### How should this be manually tested?
Implement `customViewForEmptyState()` returning a custom view.
#### What are the relevant tickets?
--
#### Questions
N/A

### CODE REVIEW

File: WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift [hunk 1]
```diff
@@ -44,17 +44,22 @@ extension UITableView: WLEmptyStateProtocol {
         }
     }
     
-    
     @objc private dynamic func swizzledReload() {
         self.swizzledReload()
```
Comment (enigma2006x) [inline WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift]: Maybe you could remove the "self"

File: WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift [hunk 2]
```diff
@@ -44,17 +44,22 @@ extension UITableView: WLEmptyStateProtocol {
...
         
         if numberOfItems == 0 && self.subviews.count > 1 {
```
Comment (enigma2006x) [inline WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift]: Maybe you could remove the "self"

File: WLEmptyState/Classes/Protocols/WLEmptyStateDataSource.swift [hunk 3]
```diff
@@ -49,4 +54,9 @@ public extension WLEmptyStateDataSource {
         let description = NSAttributedString(string: "Oops There's no data.", attributes: [NSAttributedString.Key.font: UIFont.preferredFont(forTextStyle: .caption1)])
```
Comment (enigma2006x) [inline WLEmptyState/Classes/Protocols/WLEmptyStateDataSource.swift]: Maybe you could create a struct or enum with all the Strings

**Comment by lojals** [comment]
@squallone the examples will be added in a subsequent task, refactoring the whole set of demos and adding more UITests.

**Pull request merged**